packages/core/content-manager/server/src/history/services/lifecycles.ts

```diff
@@ -172,17 +172,25 @@ const createLifecyclesService = ({ strapi }: { strapi: Core.Strapi }) => {
       });
 
       // Schedule a job to delete expired history versions every day at midnight
-      state.deleteExpiredJob = scheduleJob('0 0 * * *', () => {
+      state.deleteExpiredJob = scheduleJob('historyDaily', '0 0 * * *', () => {
         const retentionDaysInMilliseconds = serviceUtils.getRetentionDays() * 24 * 60 * 60 * 1000;
         const expirationDate = new Date(Date.now() - retentionDaysInMilliseconds);
 
-        strapi.db.query(HISTORY_VERSION_UID).deleteMany({
-          where: {
-            created_at: {
-              $lt: expirationDate.toISOString(),
+        strapi.db
+          .query(HISTORY_VERSION_UID)
+          .deleteMany({
+            where: {
+              created_at: {
+                // Only keep YYYY-MM-DD from the string because mysql doesn't support ISO 8601
+                $lt: expirationDate.toISOString().split('T').at(0),
```

Comment:
This is basically the fix. There may be a more elegant one. @Marc-Roig do you think the db engine should be the one to handle that?

Reply:
Agreed, I also worry introducing timezones like that could cause some subtle bugs unless we have tests explicitly verifying they'll work correctly

Reply:
@Marc-Roig you're right Knex handles the date object directly. I tested manually with mysql postgres and sqlite, so I updated the code